bcmc/src/main/java/com/adyen/checkout/bcmc/BcmcComponent.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by arman on 18/9/2019.
+ */
+package com.adyen.checkout.bcmc
+
+import androidx.lifecycle.viewModelScope
+import com.adyen.checkout.card.CardValidationUtils
+import com.adyen.checkout.card.data.CardType
+import com.adyen.checkout.card.data.ExpiryDate
+import com.adyen.checkout.card.repository.PublicKeyRepository
+import com.adyen.checkout.components.GenericComponentState
+import com.adyen.checkout.components.PaymentComponentProvider
+import com.adyen.checkout.components.base.BasePaymentComponent
+import com.adyen.checkout.components.base.GenericPaymentMethodDelegate
+import com.adyen.checkout.components.model.payments.request.CardPaymentMethod
+import com.adyen.checkout.components.model.payments.request.PaymentComponentData
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.components.validation.ValidatedField
+import com.adyen.checkout.core.exception.ComponentException
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import com.adyen.checkout.cse.CardEncrypter
+import com.adyen.checkout.cse.UnencryptedCard
+import com.adyen.checkout.cse.exception.EncryptionException
+import kotlinx.coroutines.launch
+
+private val TAG = LogUtil.getTag()
+
+private val PAYMENT_METHOD_TYPES = arrayOf(PaymentMethodTypes.BCMC)
+
+/**
+ * Constructs a [BcmcComponent] object.
+ *
+ * @param paymentMethodDelegate [PaymentMethodDelegate] represents payment method.
+ * @param configuration [BcmcConfiguration].
+ */
+class BcmcComponent(paymentMethodDelegate: GenericPaymentMethodDelegate, configuration: BcmcConfiguration) :
+    BasePaymentComponent<BcmcConfiguration, BcmcInputData, BcmcOutputData,
+        GenericComponentState<CardPaymentMethod>>(paymentMethodDelegate, configuration) {
+
+    companion object {
+        @JvmField
+        val PROVIDER: PaymentComponentProvider<BcmcComponent, BcmcConfiguration> = BcmcComponentProvider()
+
+        @JvmField
+        val SUPPORTED_CARD_TYPE = CardType.BCMC
+    }
+
+    private val publicKeyRepository: PublicKeyRepository = PublicKeyRepository()
```

Comment:
should we pass the repository in the constructor to allow for manual DI and testing?

Reply:
Fixed 👍